activerecord/test/cases/strict_loading_test.rb

```diff
@@ -672,6 +673,14 @@ def test_strict_loading_violation_logs_on_polymorphic_relation
     ActiveRecord.action_on_strict_loading_violation = old_value
   end
 
+  def test_strict_loading_in_n_plus_one_only_mode_does_not_eager_load_child_associations
+    developer = Developer.first
+    developer.strict_loading!(mode: :n_plus_one_only)
+    developer.projects.first
+
+    assert_not_predicate developer.projects, :loaded?
```

Comment:
I hope I'm understanding it correctly, but we can assert that calling firm on the project won't raise anymore now?
Where it did previously.
```suggestion
    assert_not_predicate developer.projects, :loaded?
    
    assert_nothing_raised do
      developer.projects.first.firm
    end
```

Reply:
That's correct. Committed suggestion